var/spack/repos/builtin/packages/mpifileutils/package.py

```diff
@@ -47,11 +50,58 @@ class Mpifileutils(AutotoolsPackage):
 
     variant('experimental', default=False,
         description="Install experimental tools")
-
-    # --enable-experimental fails with v0.6 and earlier
     conflicts('+experimental', when='@:0.6')
 
-    def configure_args(self):
+    variant('gpfs', default=False,
+        description="Enable optimizations and features for GPFS")
+    conflicts('+gpfs', when='@:0.8.1')
+
+    def configure_args_cmake(self):
+        args = []
+        args.append("-DWITH_DTCMP_PREFIX=%s" % self.spec['dtcmp'].prefix)
+        args.append("-DWITH_LibCircle_PREFIX=%s" %
+                    self.spec['libcircle'].prefix)
+
+        if '+xattr' in self.spec:
+            args.append("-DENABLE_XATTRS=ON")
+        else:
+            args.append("-DENABLE_XATTRS=OFF")
+
+        if '+lustre' in self.spec:
+            args.append("-DENABLE_LUSTRE=ON")
+        else:
+            args.append("-DENABLE_LUSTRE=OFF")
+
+        if '+gpfs' in self.spec:
+            args.append("-DENABLE_GPFS=ON")
+        else:
+            args.append("-DENABLE_GPFS=OFF")
+
+        if '+experimental' in self.spec:
+            args.append("-DENABLE_EXPERIMENTAL=ON")
+        else:
+            args.append("-DENABLE_EXPERIMENTAL=OFF")
+
+        return args
+
+    @when('@0.9:')
+    def install(self, spec, prefix):
+        args = []
+        args.append('-DCMAKE_INSTALL_PREFIX=%s' % prefix)
```

Comment:
This would also be cleaner if you moved all of the argument generation into `cmake_args`, and similarly for `configure_args` and the install method below.

The prefix is available in `cmake_args` as `self.spec.prefix`, so even though it's not explicitly passed you have access to it all the same.